onvm/onvm_mgr/onvm_nf.c

```diff
@@ -122,19 +123,24 @@ onvm_nf_check_status(void) {
                         case MSG_NF_STARTING:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
                                 if (onvm_nf_start(nf) == 0) {
-                                        onvm_stats_add_event("NF Starting", nf);
+                                        onvm_stats_gen_event_nf_info("NF Starting", nf);
                                 }
                                 break;
                         case MSG_NF_READY:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
                                 if (onvm_nf_ready(nf) == 0) {
-                                        onvm_stats_add_event("NF Ready", nf);
+                                        onvm_stats_gen_event_nf_info("NF Ready", nf);
                                 }
                                 break;
                         case MSG_NF_STOPPING:
                                 nf = (struct onvm_nf_info *)msg->msg_data;
+                                if (nf == NULL)
+                                        break;
+
+                                /* Get instance_id before tearing down nf */
+                                stop_nf_id = (void *)&(nf->instance_id);
```

Comment:
Yeah we can't do this, the assumption is the nf memory can be freed, thus we need to save instance_id in a temp value, you're pointing to a value which doesn't quite cut it as it might get freed.

Instead just try doing
```
/* Saved as onvm_nf_stop frees the memory*/
uint16_t stop_nf_id = (nf->instance_id);
onvm_stats_gen_event_info("NF Stopping", ONVM_EVENT_NF_STOP, &stop_nf_id);
```

I'm assuming that will work although double check